framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl.java

```diff
@@ -0,0 +1,194 @@
+import org.kie.cloud.api.deployment.Deployment;
+import org.kie.cloud.api.deployment.KieServerDeployment;
+import org.kie.cloud.api.deployment.SmartRouterDeployment;
+import org.kie.cloud.api.deployment.WorkbenchDeployment;
+import org.kie.cloud.api.deployment.constants.DeploymentConstants;
+import org.kie.cloud.api.scenario.ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenario;
+import org.kie.cloud.common.provider.KieServerControllerClientProvider;
+import org.kie.cloud.openshift.constants.OpenShiftConstants;
+import org.kie.cloud.openshift.constants.OpenShiftTemplateConstants;
+import org.kie.cloud.openshift.deployment.DatabaseDeploymentImpl;
+import org.kie.cloud.openshift.deployment.KieServerDeploymentImpl;
+import org.kie.cloud.openshift.deployment.SmartRouterDeploymentImpl;
+import org.kie.cloud.openshift.deployment.WorkbenchRuntimeDeploymentImpl;
+import org.kie.cloud.openshift.resource.Project;
+import org.kie.cloud.openshift.template.OpenShiftTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl extends OpenShiftScenario implements ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenario {
+
+    private WorkbenchDeployment workbenchRuntimeDeployment;
+    private SmartRouterDeployment smartRouterDeployment;
+    private KieServerDeploymentImpl kieServerOneDeployment;
+    private KieServerDeploymentImpl kieServerTwoDeployment;
+    private DatabaseDeploymentImpl databaseOneDeployment;
+    private DatabaseDeploymentImpl databaseTwoDeployment;
+
+    private Map<String, String> envVariables;
+
+    private static final Logger logger = LoggerFactory.getLogger(ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl.class);
+
+    public ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl(Map<String, String> envVariables) {
+        this.envVariables = envVariables;
+    }
+
+    @Override
+    public void deploy() {
+        super.deploy();
+
+        logger.info("Processing template and creating resources from " + OpenShiftTemplate.CLUSTERED_CONSOLE_SMARTROUTER_TWO_KIE_SERVERS_TWO_DATABASES.getTemplateUrl().toString());
+        envVariables.put(OpenShiftTemplateConstants.IMAGE_STREAM_NAMESPACE, projectName);
+        project.processTemplateAndCreateResources(OpenShiftTemplate.CLUSTERED_CONSOLE_SMARTROUTER_TWO_KIE_SERVERS_TWO_DATABASES.getTemplateUrl(), envVariables);
+
+        workbenchRuntimeDeployment = createWorkbenchRuntimeDeployment(project);
+        workbenchRuntimeDeployment.scale(1);
+
+        smartRouterDeployment = createSmartRouterDeployment(project);
+        smartRouterDeployment.scale(1);
+
+        kieServerOneDeployment = createKieServerDeployment(project, "1");
+        kieServerOneDeployment.scale(1);
+
+        kieServerTwoDeployment = createKieServerDeployment(project, "2");
+        kieServerTwoDeployment.scale(1);
```

Comment:
Why scaling to 1? If everything is scaled to one, then this "Clustered" scenario is same as WorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenario, is it?

Reply:
Yes, it would be basically the same.
The scaling to 1 is used to unify the state of every scenario so tests already know the scenario state and can rely on it regardless of deployed scenario.
It means that when the test starts it knows that there is just one Kie server, one Workbench... so the behaviour is same for all scenarios. It allows us to reuse same tests for many different scenarios.

Reply:
On the second thought, is a test relevant to the scenario in case it needs to scaled first? Will think about it.